src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -553,4 +551,81 @@ public void refreshStudyInformation() {
             return ImmutableSetMultimap.of();
         }
     }
+
+    @Override
+    public UUID getParticipantEntityKeyId( String participantId, UUID participantsEntitySetId, SearchApi searchApi ) {
+        DataSearchResult result = searchApi.executeEntitySetDataQuery(
+                participantsEntitySetId,
+                new SearchTerm( participantIdPropertyTypeId.toString() + ":\"" + participantId + "\"", 0, 1 )
+        );
+        if ( result.getHits().size() != 1 ) {
+            return null;
+        }
+        SetMultimap<FullQualifiedName, Object> data = result.getHits().iterator().next();
+        if ( data.containsKey( INTERNAL_ID_FQN ) ) {
+            return UUID.fromString( data.get( INTERNAL_ID_FQN ).iterator().next().toString() );
+        }
+        return null;
+    }
+
+    @Override
+    public ParticipationStatus getParticipationStatus( UUID studyId, String participantId ) {
+
+        EdmApi edmApi;
+        SearchApi searchApi;
+
+        try {
+            ApiClient apiClient = apiClientCache.get( ApiClient.class );
+            edmApi = apiClient.getEdmApi();
+            searchApi = apiClient.getSearchApi();
+        } catch ( ExecutionException e ) {
+            logger.error( "Unable to load apis." );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        String participantsEntitySetName = ChronicleServerUtil.getParticipantEntitySetName( studyId );
+        UUID participantsEntitySetId = edmApi.getEntitySetId( participantsEntitySetName );
+        if ( participantsEntitySetId == null ) {
+            logger.error( "unable to get the participants EntitySet id for studyId = {}", studyId );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        UUID participantEntityKeyId = getParticipantEntityKeyId( participantId, participantsEntitySetId, searchApi );
+        if ( participantEntityKeyId == null ) {
+            logger.error( "unable to load participant entity key id for participantId = {}", participantId );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        Map<UUID, List<NeighborEntityDetails>> neighborResults = searchApi.executeFilteredEntityNeighborSearch(
+                participantsEntitySetId,
+                new EntityNeighborsFilter(
+                        Sets.newHashSet( participantEntityKeyId ),
+                        java.util.Optional.of( Sets.newHashSet( participantsEntitySetId ) ),
```

Comment:
do participants have neighbors that are participants?

Reply:
^it looks like you're filtering out any non-study neighbors, so this should probably be an empty set

Reply:
I don't think participants have neighbors that are participants. The goal is to get neighbors that match participants -> participated in -> study. Are you saying that the sourceEntitySetIds should be empty?

I was curious about this because the entitySetId param is "participantsEntitySetId", and I wasn't sure if it made a difference if sourceEntitySetIds was `[]` or `[participantsEntitySetId]`.

Reply:
it should be blank -- the src / dst / association entity set ids are relative to the entityKeyIds that you pass in. So if you pass in a set of participant entityKeyIds (for simplicity let's say the ids are 1, 2, and 3) and include `participantsEntitySetId` as a src filter, `studyEntitySetId` as a dst filter, and `participatedInEntitySetId` as an association filter, then the neighbors response will include all entity triplets of any of these forms:

`<participant entity> -> <participated in entity> -> 1`
`<participant entity> -> <participated in entity> -> 2`
`<participant entity> -> <participated in entity> -> 3`
`1 -> <participated in entity> -> <study entity>`
`2 -> <participated in entity> -> <study entity>`
`3 -> <participated in entity> -> <study entity>`